biz.aQute.bnd.maven/src/aQute/bnd/maven/lib/resolve/BndrunContainer.java

```diff
@@ -262,9 +263,19 @@ public void setRunrequiresFromProjectArtifact(Run run) {
 			&& (project.getPlugin("biz.aQute.bnd:bnd-maven-plugin") != null)) {
 
 			Artifact artifact = project.getArtifact();
+			String bsn = artifact.getArtifactId();
+
+			try (Jar jar = new Jar(artifact.getFile())) {
```

Comment:
This is kind of expensive as it will read the entire jar when all you need is the manifest. It is less costly to use Bnd Domain class (`Domain.domain(File)`) to process the manifest and get the BSN.

https://github.com/bndtools/bnd/blob/aea111f779c5dfd2a91a82d698abb28bb25aaf0b/biz.aQute.bndlib/src/aQute/bnd/osgi/Domain.java#L509-L519

Reply:
right right, I'll fix.

Reply:
fixed